internal/aws/cwlogs/cwlog_client.go

```diff
@@ -34,6 +38,17 @@ type Client struct {
 	tags         map[string]*string
 	logger       *zap.Logger
 }
+type UserAgentOption func(*UserAgentFlag)
+
+type UserAgentFlag struct {
+	isAppSignals bool
+}
```

Comment:
```suggestion
type cwLogClient struct {
	userAgentPostFix []string
}
```

I don't think this needs to be exported and it could instead be a list of strings that gets appended to the user agent header. Name could be better but I'm out of ideas atm.

Reply:
Updated using your suggestion with names: `cwLogClientConfig` and `userAgentExtras`